powersimdata/data_access/launcher.py

```diff
@@ -0,0 +1,202 @@
+import posixpath
+import sys
+
+import requests
+
+from powersimdata.utility import server_setup
+
+
+def _check_threads(threads):
+    """Validate threads argument
+
+    :param int threads: the number of threads to be used
+    :raises TypeError: if threads is not an int
+    :raises ValueError: if threads is not a positive value
+    """
+    if threads:
+        if not isinstance(threads, int):
+            raise TypeError("threads must be an int")
+        if threads < 1:
+            raise ValueError("threads must be a positive value")
+
+
+def _check_solver(solver):
+    """Validate solver argument
+
+    :param str solver: the solver used for the optimization
+    :raises ValueError: if invalid solver provided
+    """
+    solvers = ("gurobi", "glpk")
+    if solver is not None and solver.lower() not in solvers:
+        raise ValueError(f"Invalid solver: options are {solvers}")
+
+
+class Launcher:
```

Comment:
Missing docstring. And, usually, we add here the parameter (here `scenario`) that is needed to instantiate the class